src/EventStore.Core/Services/PersistentSubscription/PersistentSubscriptionService.cs

```diff
@@ -612,8 +647,22 @@ string user
 					_checkpointReader,
 					new PersistentSubscriptionCheckpointWriter(key, _ioDispatcher),
 					new PersistentSubscriptionMessageParker(key, _ioDispatcher)));
+			
+			if (_subscriptionsById.ContainsKey(key)) {
```

Comment:
suggestion: move this check before creating the PersistentSubscription object, otherwise it's not used.
it should also be ok to remove the Log.Information line since we don't currently log these errors:

```
			if (_subscriptionsById.ContainsKey(key))
				return false;
```

Reply:
Made the changes

Reply:
thanks!